server/controllers/tournament/createTournament.ts

```diff
@@ -0,0 +1,10 @@
+import catchAsync from '../../utils/catchAsync';
+
+const createTournament = catchAsync(
+  (req, res, next) => {
+    // waiting for database models
+    res.status(204).end();
+  },
+);
```

Comment:
I see that you've still kept the request handler in a separate file. That makes me think you want to put the whole use-case in here which IMO is not a good idea. My reason for such thinking is that in one unit (be it function or a file) you will mix business logic and HTTP related stuff.

What I'd suggest is to put the handler and all the HTTP related code in the router and make it call a function defined in a different file that does just the business logic (auth, validations and persistence).

Reply:
@eastemployeer What do you think? You told me about this conception.
I think you mean our first version of the concept which was something like that:
For example @mblaze You want to return object  like  `{ httpStatus, message, body }` from controller to router where returned object will be passed to the response?